### ISSUE TITLE
Introduce test ID

### ISSUE THREAD

**Issue by f3ath** [jsonpath-standard/jsonpath-compliance-test-suite#36]
Some implementations may want to ignore specific tests, e.g. due to hard limitation of the underlying platforms. For such cases we might want to introduce a test ID. Such an ID should be stable, i.e. it should not change when the test metadata (like name) changes, neither it should depend on the order of the results.

I am considering adding the `id` field to the test object:
```json
    {
      "name": "name shorthand",
      "id": "17e682f060b5f8e47ea04c5c4855908b0a5ad612022260fe50e11ecb0cc0ab76",
      "selector": "$.a",
      "document": {
        "a": "A",
        "b": "B"
      },
      "result": [
        "A"
      ]
```

The `id` is calculated as follows:
1. take the `test` object
2. remove the `name` key
3. sort the `result` array if it exists (this should not affect the semantics since the traversal order is explicitly undefined)
4. apply JSON encode
5. take `sha256` hash sum of the encoded string

@gregsdennis @glyn let me know what you think.

**Comment by gregsdennis** [comment]
I just use the test name.

https://github.com/gregsdennis/json-everything/blob/master/JsonPath.Tests/Suite/ComplianceTestSuiteTests.cs#L15-L17

I don't think we have anything in the CI that guarantees uniqueness, though.  We'd have to put something in whether we use the name or an id.

**Comment by glyn** [comment]
My general view of this change is that it's going to make maintenance more difficult. Better to simply ensure that all the test case names are unique - something that could be checked when cts.json is generated.
 
>     3. sort the `result` array if it exists (this should not affect the semantics since the traversal order is explicitly undefined)

The order of the result array is partially defined by the spec. For instance [*] applied to an array produces the elements of the array in array order.

That said we still need to address issue #1, which may further complicate matters.

**Comment by f3ath** [comment]
This is partially solved by #92. I don't see much value in keeping this open.
